oauth2_provider/views.py

```diff
@@ -58,10 +63,43 @@ class AccessTokenView(provider.oauth2.views.AccessTokenView):
     and available to the claim handlers configured by `OAUTH_OIDC_ID_TOKEN_HANDLERS`
 
     """
+    grant_types = provider.oauth2.views.AccessTokenView.grant_types + ["edx_session"]
 
     # Add custom authentication provider, to support email as username.
     authentication = (provider.oauth2.views.AccessTokenView.authentication +
-                      (PublicPasswordBackend, ))
+                      (PublicPasswordBackend, ExistingSessionBackend))
+
+    def edx_session(self, request, data, client):
+        """
+        Handle ``grant_type=edx_session`` requests.
+        """
+        grant_data = self.get_edx_session_grant(request, request.POST, client)
+        at = self.create_access_token(request, grant_data['user'], grant_data['scope'], client)
```

Comment:
Ignore me, I see what's going on. `access_token_response_data()` injects the `id_token` into the response.